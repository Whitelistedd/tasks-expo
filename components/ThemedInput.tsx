import {
  StyleSheet,
  TextInput,
  TextInputProps,
  View,
  type ViewProps,
} from "react-native";

import { useThemeColor } from "@/hooks/useThemeColor";
import { ThemedText } from "./ThemedText";
import React, { ForwardedRef } from "react";

export type ThemedInputProps = TextInputProps & {
  lightColor?: string;
  darkColor?: string;
  label?: string;
};

export const ThemedInput = React.forwardRef(function ThemedInput(
  { style, lightColor, darkColor, label, ...otherProps }: ThemedInputProps,
  ref: ForwardedRef<TextInput>
) {
  const inputColor = useThemeColor(
    { light: lightColor, dark: darkColor },
    "input"
  );

  return (
    <View style={styles.container}>
      {label && <ThemedText type="desc">{label}</ThemedText>}
      <TextInput
        ref={ref}
        style={[{ borderColor: inputColor }, styles.input, style]}
        {...otherProps}
      />
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    gap: 5,
  },
  input: {
    borderWidth: 1,
    flex: 1,
    borderRadius: 8,
    padding: 8,
  },
});
