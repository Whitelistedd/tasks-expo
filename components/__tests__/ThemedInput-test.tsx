import { fireEvent, render } from "@testing-library/react-native";
import { ThemedInput } from "../ThemedInput";

describe("Button Component", () => {
  it("renders correctly", () => {
    const tree = render(<ThemedInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("given label, should match snapshot render correctly", () => {
    const tree = render(<ThemedInput label="Test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("given label, should render it correctly", () => {
    const { getByText } = render(<ThemedInput label="Test" />);

    const label = getByText("Test");

    expect(label).toBeTruthy();
  });

  it("given value, contains correct given value", () => {
    const { getByTestId } = render(
      <ThemedInput value="Hello" testID="input" />
    );

    const input = getByTestId("input");
    expect(input.props.value).toBe("Hello");
  });

  it("on input text change, should update the input value", () => {
    const handleTextChange = jest.fn();

    const { getByTestId } = render(
      <ThemedInput onChangeText={handleTextChange} testID="input" />
    );

    const input = getByTestId("input");

    fireEvent.changeText(input, "Hello Test");

    expect(handleTextChange).toHaveBeenCalledWith("Hello Test");
  });
});
