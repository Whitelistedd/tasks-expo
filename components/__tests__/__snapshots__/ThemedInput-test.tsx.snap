// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component given label, should match snapshot render correctly 1`] = `
<View
  style={
    {
      "gap": 5,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 13,
          "fontWeight": "500",
        },
        undefined,
      ]
    }
  >
    Test
  </Text>
  <TextInput
    style={
      [
        {
          "borderColor": "#11181C",
        },
        {
          "borderRadius": 8,
          "borderWidth": 1,
          "flex": 1,
          "padding": 8,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Button Component renders correctly 1`] = `
<View
  style={
    {
      "gap": 5,
    }
  }
>
  <TextInput
    style={
      [
        {
          "borderColor": "#11181C",
        },
        {
          "borderRadius": 8,
          "borderWidth": 1,
          "flex": 1,
          "padding": 8,
        },
        undefined,
      ]
    }
  />
</View>
`;
